### DISCUSSION TITLE
How it really works

### DISCUSSION THREAD

**Discussion by BitDownMod** [markqvist/Reticulum#475] [General]
Sorry for the noob question.

I have two Androids phones and two TTGO LoRa boards
![IMG_5668](https://github.com/markqvist/Reticulum/assets/30846278/3cb2c8b5-1376-4e25-99ed-33db0473d938)

If I install the Sideband app to these phones and connect the LoRa board (RNode installed) will I be able to establish a link between them? Or do I need something else, such as a special node?

**Comment by Erethon** [comment]
Yes, you'll be able to establish a link between the two phones with this setup. You'll need to pair the two rnodes with the phones via bluetooth first and set up the relevant LoRa parameters (frequency, coding rate, bandwidth, spreading factor) and then it will work, no further hardware required.

**Comment by BitDownMod** [comment]
Thanks! And if I want a repeater, I just need one more board, right? Should it be configured differently?

**Reply by Erethon** [reply]
In Reticulum the LoRa boards only act as radio interfaces (for now at least). This means you'll need one more LoRa board **and** some sort of device that's gonna run the Reticulum stack. This can be a single board computer (i.e. a Raspberry Pi), a computer (laptop/desktop) or another phone running Sideband.

The "repeater" [1] node should be set as a [Transport node](https://markqvist.github.io/Reticulum/manual/understanding.html#node-types) (or in Sideband, Settings -> Connectivity -> Enable Reticulum Transport).

[1]: Repeater is in quotes because in Reticulum you can't have dumb repeaters that only repeat signals/packets. Every node in a network is a peer and knows how to route packets.

**Reply by joakim** [reply]
I'll just add that the LoRa board can't be UART based, as some of the cheaper ones are.

As Mark told me on the Matrix room:

> UART will not work, it's too slow to do fast and reliable channel activity detection. The RNode firmware needs the direct SPI bus to the modem, and access to the interrupt pins as well.

**Reply by markqvist** [reply]
Just to clarify on this, we're talking about the connection to the *modem IC*. You can definitely connect a host system to an RNode over UART, but the RNode itself needs an SPI connection and interrupt lines to the modem chip that is sitting on it, since it needs to poll the transceiver about 300 times per second to get an accurate assessment of the real-time channel characteristics.

The whole point of having the RNode firmware is letting it handle all the heavy lifting related to *physical radio medium*. Then Reticulum can handle the rest :)

So to use Reticulum over LoRa, you need to get one of the many different supported boards that have a microprocessor capable of running the RNode firmware, and a supported LoRa modem chip, connected over SPI. You can of course also build your own, completely custom setup, as long as it uses this methodology. But by far, the easiest way to get started is just to use one of the supported boards, flash the RNode firmware to it, and connect it to whatever you're running Reticulum on.

**Reply by joakim** [reply]
Thanks for the clarification :) I'm still learning about all this, so that was useful.

**Reply by F4JTC** [reply]
Bonjour à tous, je re prend cette discution car je voudrais faire des modules répéteur autonome, ou digipeteur comme disent certain, existe t il maintenant ( septembre 2025) un firmware qui ferait fonctionner les modules sans ajout de carte extérieure, raspberry pi ou autre ? ou cela n'existera t  il jamais du fait que le " bus " des lilogo/heltec est trop faible ?
Vous comprendrais le but de ma question ; le prix serait considérablement réduit sans raspeberry et le déploiement serait grandement facilité - 
merci !

Hello everyone, I'm resuming this discussion because I'd like to make standalone repeater modules, or Digipeter modules as some say. Is there now (September 2025) firmware that would make the modules work without adding an external card, Raspberry Pi or something else? Or will it never exist because the lilogo/heltec "bus" is too weak?
You understand the point of my question; the price would be considerably reduced without a Raspberry Pi, and deployment would be greatly facilitated.
Thank you!
